plugins/woocommerce/includes/admin/settings/class-wc-settings-accounts.php

```diff
@@ -237,6 +239,23 @@ protected function get_settings_for_default_section() {
 			),
 		);
 
+		// Hide username setting when using the block based checkout.
+		if ( CartCheckoutUtils::is_checkout_block_default() ) {
+			$account_settings = array_map(
+				function ( $setting ) {
+					switch ( $setting['id'] ) {
+						case 'woocommerce_registration_generate_username':
+							return array();
```

Comment:
Seems a bit odd leaving an empty array hanging, why not using the good ol' `foreach` instead of `array_map()` to unset this key?

I am approving nonetheless as this is a nitpick, good job 👍🏼

Reply:
We can use `array_filter`.

Reply:
https://github.com/woocommerce/woocommerce/pull/49389/commits/3e625203ced081c6cc6e40d52a3c4bb4ef71865f